### ISSUE TITLE
RigidBodies clip/sink into static geometry at slightly high speeds

### ISSUE THREAD

**Issue by Gonkalbell** [jcornaz/heron#276]
I first noticed this issue in [bevy_ecs_ldtk](https://github.com/Trouv/bevy_ecs_ldtk)'s platformer example, where the player clips through the ground and slowly rises back up:

![Screencast from 06-05-2022 07 48 48 PM](https://user-images.githubusercontent.com/5430340/172076144-26c6988f-cec5-4e47-bab2-3bac7e6e716d.gif)

It looks like I can repro the same issue in this crate's examples as well. Here I just added a wall to the `events` example, and tried running into it. The problem is noticeable with the default `const SPEED: f32 = 300.0;`, but changing the speed to something higher like 600 makes it much easier to see:

![Screencast from 06-05-2022 07 59 21 PM](https://user-images.githubusercontent.com/5430340/172076279-1a5bb484-43c6-48ff-b696-4f44fc3ae90c.gif)

My guess is it's just an issue with the default values for `PhysicsSteps`, or perhaps something like `IntegrationParameters::max_position_iterations` is to small by default.

**Comment by jcornaz** [comment]
Isn't that rapier's behavior?

**Comment by Gonkalbell** [comment]
It may be. I tried reproing this in rapier_2d's examples. I believe I did still see some clipping, but it wasn't quite as extreme. Maybe there's a difference in the default settings? I could open an issue in reapier's repo, though, if that's where you believe the bug is.

**Comment by jcornaz** [comment]
I am not sure it is even a bug at all. I think it is rapier's *intended* behavior.

But that behavior is likely to be impacted by the step rate and whether ccd is enabled.

May I ask what is your FPS? And maybe can you try [with CCD enabled](https://github.com/jcornaz/heron/issues/199#issuecomment-1052453851)?

**Comment by Gonkalbell** [comment]
`FrameTimeDiagnosticsPlugin` is reporting a consistent 60fps. I think the screen captures I took were at a lower framerate, so that's why the gifs I posted look janky.

I tried out enabling CCD, but that didn't seem to help. I also tried setting `IntegrationParameters::max_position_iterations` to something like 4 or 8. There still seemed to be some collider penetration, but it didn't seem to take as many frames to exit the other collider.

You're probably right that this is just an issue in rapier. Rapier's changlog seems to mention changes to integration calculation and resolving collider penetration in versions 0.12 and 0.13, so I'll check if it's resolved upstream.

**Comment by jcornaz** [comment]
Ok. Thanks for the info.

I do plan to update the version of rapier used by heron. But I would like to batch that update with the next update of bevy so that I don't break the API the rapier update. 

Though, if updating rapier does fix that kind of problem, I may consider updating it before the next bevy release comes out.

**Comment by jcornaz** [comment]
@Gonkalbell [heron 4.0.0-alpha.1](https://github.com/jcornaz/heron/releases/tag/v4.0.0-alpha.1) is using rapier 0.13.0. Can you try with that version and tell me if the problem persists?

**Comment by bit-garden** [comment]
Tldr; Mathmatically speaking, your objects are too big and more gravity doesn't fix it. 

Lower the gravity and use smaller objects to fix the simulation, then scale up your rendering to look right on the sceen. (PS, Bevy_rapier has a [solution](https://rapier.rs/docs/user_guides/bevy_plugin/common_mistakes/#why-is-everything-moving-in-slow-motion) for this.)

---

The speeds are supposed to be in meters. But Bevy_rapier provides a way to set a pixel to meter ratio.

In my initial attempts to use Heron, I had a dynamic ball at `0, 0` and a static block at `0, -200` with gravity at `0. -9.8`. The simulation appeared slow and the ball would slowly correct itself on impact. I don't see a scaling option currently for Heron.

I did the same setup in Bevy_rapier and it yields more expected results of stopping on impact instead of falling into the block, but I scaled 100 pixels to a meter(effectively 100th the size).

[This](https://github.com/Trouv/bevy_ecs_ldtk/blob/0d3505c6ef09cffb56995476ce59e6c7ff15770b/examples/platformer/main.rs#L17) also indicates that the [bevy_ecs_ldtk](https://github.com/Trouv/bevy_ecs_ldtk)'s platformer example is not working with trivial amounts of gravity.

```
.insert_resource(Gravity::from(Vec3::new(0.0, -2000., 0.0)))
```
accelerating 2000 meters(4473 miles per hour) per second is a lot. 

Reference:
[Common mistakes](https://rapier.rs/docs/user_guides/bevy_plugin/common_mistakes/#why-is-everything-moving-in-slow-motion)

> The recommended way of using Rapier is to use [SI units](https://en.wikipedia.org/wiki/International_System_of_Units) (meters, seconds, kilograms, etc.) If the player sprite is a 100x100 cuboid, then it is as if your player is 100 meters tall and 100 meters wide, which is huge. Therefore it is recommended ot have a scaling factor between the graphics and the physics. For example we can say that 1 physics meter is equal to 50 pixels. This means that we can initialize our player collider as a 2x2 cuboid while still using a 100x100 pixels sprite.
> 
> In this example, we could set RapierPhysicsPlugin::pixels_per_meter(50.0) so that while all the transforms, collider size, positions, etc. are expressed in pixels on your end, bevy_rapier will automatically divide them by 50.0 whenever it inputs/reads data into/from Rapier itself.

**Comment by jcornaz** [comment]
Yes, that's right. Thanks @bit-garden for pointing this out.

I am closing this issue in favor of #284
